spec/models/sdg/target_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "rails_helper"
+
+describe SDG::Target do
+  it "is valid" do
+    expect(build(:sdg_target, code: "1.Z", goal: SDG::Goal.find_by(code: "1"))).to be_valid
+  end
+
+  it "is not valid without a code" do
+    expect(build(:sdg_target, code: nil)).not_to be_valid
+  end
+
+  it "is not valid without a goal" do
+    target = build(:sdg_target, goal: nil)
+
+    expect(target).not_to be_valid
+  end
+
+  it "is not valid when code is duplicated for the same parent goal" do
```

Comment:
I think we have to update this test: it is not valid even if the parent goal is not the same.